Assets/Scripts/Player/Weapon/Weapon.cs

```diff
@@ -1,18 +1,86 @@
 using System.Collections;
+using Unity.VisualScripting;
 using UnityEngine;
-public abstract class Weapon : MonoBehaviour
+using UnityEngine.InputSystem;
+using UnityEngine.Pool;
+
+public class Weapon : MonoBehaviour
 {
-    public string weaponName { get; set; }
-    public int attackDamage { get; set; }
-    public float reloadTime { get; set; }
-    public float fireDelay { get; set; }
-    public float bulletSpeed { get; set; }
-    public string bulletPrefabPaths { get; set; }
-    public string gunPrefabPath { get; set; }
-    public string soundFXPath { get; set; }
-    public string visualFXPath { get; set; }
-    public int method {  get; set; }
-    public virtual void Init() { }
-    public virtual void Fire() { }
-    public virtual void Reload() { }
+    [Header ("DataField")]
+    [SerializeField] private WeaponData weaponData;
+    [SerializeField] WaitForSeconds waitForFireDelay;
+
+    [Header("TestField")]
+    public GameObject bulletPrefabTest;
+
+    [SerializeField] Transform player;
+    [SerializeField] private Animator animator;
+    [SerializeField] Transform gunMuzzle;
+    [SerializeField] Transform rayCastTarget;
+    [SerializeField] private bool isShootReady = false;
+    [SerializeField] private bool isHitScan = false;
+
+    private void Awake()
+    {
+        player = Camera.main.transform;
+        animator = GetComponent<Animator>();
+    }
+
+
+    public void Initialize(WeaponData weapon)
+    {
+        weaponData = weapon;
+        bulletPrefabTest = Resources.Load<GameObject>(weaponData.bulletPrefab);
+        waitForFireDelay = new WaitForSeconds(weaponData.fireDelay);
+        isShootReady = true;
+        if(weaponData.method == 1)
+        {
+            isHitScan = true;
+        }
+    }
+
+    private void Update()
+    {
+        if (Mouse.current.leftButton.wasPressedThisFrame && isShootReady) 
+        {
+            Fire();
+        }
+    }
+
+    public void Fire()
+    {
+        StartCoroutine(FireCoroutine());
```

Comment:
```suggestion
if (isShootReady)  StartCoroutine(FireCoroutine());
```

isShootReady는 Fire에서 검사해야 할 사전 조건